src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -369,6 +412,111 @@ export class InputTimePicker
   //
   // --------------------------------------------------------------------------
 
+  private delocalizeTimeString(value: string): string {
+    const locale = this.effectiveLocale.toLowerCase();
+    let localeConfig,
+      valueToParse = value;
+
+    if (locale === "ar") {
+      if (this.numberingSystem === "arab") {
+        const arabNumberMap = {
+          "١": "1",
+          "٢": "2",
+          "٣": "3",
+          "٤": "4",
+          "٥": "5",
+          "٦": "6",
+          "٧": "7",
+          "٨": "8",
+          "٩": "9",
+          "٠": "0"
+        };
+        valueToParse = value.replace(/[١٢٣٤٥٦٧٨٩٠]/g, (match) => arabNumberMap[match]);
+      } else if (this.numberingSystem === "arabext") {
+        const arabextNumberMap = {
+          "۱": "1",
+          "۲": "2",
+          "۳": "3",
+          "۴": "4",
+          "۵": "5",
+          "۶": "6",
+          "۷": "7",
+          "۸": "8",
+          "۹": "9",
+          "۰": "0"
+        };
+        valueToParse = value.replace(/[۱۲۳۴۵۶۷۸۹۰]/g, (match) => arabextNumberMap[match]);
+      }
+      localeConfig = {
+        meridiem: (hour) => (hour > 12 ? "م" : "ص"),
+        formats: {
+          LT: "HH:mm A",
+          LTS: "HH:mm:ss A",
+          L: "DD/MM/YYYY",
+          LL: "D MMMM YYYY",
+          LLL: "D MMMM YYYY HH:mm A",
+          LLLL: "dddd D MMMM YYYY HH:mm A"
+        }
+      };
+    } else if (locale === "en-au") {
+      localeConfig = {
+        meridiem: (hour) => (hour > 12 ? "pm" : "am")
+      };
+    } else if (locale === "en-ca") {
+      localeConfig = {
+        meridiem: (hour) => (hour > 12 ? "p.m." : "a.m.")
+      };
+    } else if (locale === "el") {
+      localeConfig = {
+        meridiem: (hour) => (hour > 12 ? "μ.μ." : "π.μ.")
+      };
+    } else if (locale === "hi") {
+      localeConfig = {
+        formats: {
+          LT: "h:mm A",
+          LTS: "h:mm:ss A",
+          L: "DD/MM/YYYY",
+          LL: "D MMMM YYYY",
+          LLL: "D MMMM YYYY, h:mm A",
+          LLLL: "dddd, D MMMM YYYY, h:mm A"
+        },
+        meridiem: (hour) => (hour > 12 ? "pm" : "am")
+      };
+    } else if (locale === "ko") {
+      localeConfig = {
+        meridiem: (hour) => (hour > 12 ? "오후" : "오전")
+      };
+    } else if (locale === "zh-tw") {
+      localeConfig = {
+        formats: {
+          LT: "AHH:mm",
+          LTS: "AHH:mm:ss"
+        }
+      };
+    } else if (locale === "zh-hk") {
+      localeConfig = {
+        formats: {
+          LT: "AHH:mm",
+          LTS: "AHH:mm:ss"
+        },
+        meridiem: (hour) => (hour > 12 ? "下午" : "上午")
+      };
+    }
+
+    dayjs.updateLocale(locale, localeConfig);
```

Comment:
This should be done when a locale bundle is loaded and not on every call to `delocalizeTimeString`.